certif/app/services/session.js

```diff
@@ -1,13 +1,23 @@
 import { inject as service } from '@ember/service';
 import SessionService from 'ember-simple-auth/services/session';
 
+const FRANCE_TLD = 'fr';
+const DEFAULT_LOCALE = 'fr';
+const FRENCH_INTERNATIONAL_LOCALE = 'fr';
+const FRENCH_FRANCE_LOCALE = 'fr-FR';
+
 export default class CurrentSessionService extends SessionService {
   @service currentUser;
-  @service router;
-  @service url;
+  @service currentDomain;
+  @service locale;
+  @service intl;
+  @service dayjs;
+
+  _localeFromQueryParam = undefined;
```

Comment:
Pourquoi as-tu eu besoin de la définir ici cette variable ?
De ce que je vois :
- Elle n'est utilisée que dans la fonction loadCurrentUserAndSetLocale qui la reçoit depuis le beforeModel dans application
`const localeFromQueryParam = transition.to.queryParams.lang;`
- si je fais tourner les tests de `certif/tests/unit/routes/application_test.js` localeFromQueryParam est à undefined quand elle n'est pas dans les queryParams
J'ai donc envie de supprimer cette ligne

Reply:
La variable définition de la variable `_localeFromQueryParam` n'est effectivement pas nécessaire d'un strict point de vue syntaxique, mais elle permet de facilement indiquer aux développeurs qu'un attribut de classe « privé » est disponible dans la classe. Par ailleurs les tests plantent si le bloc de code suivant est commenté et donc cette variable est à la fois utile et testée :
```javascript
    if (this._localeFromQueryParam) {
      this._setLocale(this._localeFromQueryParam);
      return;
    }
```